### PR TITLE
Repo Gardening: Consider Forum Links as Support References

### PR DESCRIPTION
**Pull request by Aurorum**
## Proposed changes:

I'd like to propose accepting WordPress.com forum links as a valid support reference. It's quite common for this to be used in issues (at least 600 cases [here](https://github.com/search?q=org%3AAutomattic+%22wordpress.com%2Fforums%22&type=issues)), but these currently aren't considered when counting interactions for escalation. 

Anecdotally, I think this is particularly problematic in the Themes repo because many themes issues are fixed with Custom CSS, and therefore, without adding the interaction to GitHub. This generally isn't possible for those using the forums (since the forums are mostly used by those on the Free plan). As such, these issues become disproportionately neglected when it would be nice for them to get some attention. 

## Jetpack product discussion

None - please feel free to close if this doesn't make sense!

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Not possible to test in this repo, but please feel free to try some links here: Aurorum/jetpack#5

### CODE REVIEW

File: projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js [hunk 1]
```diff
@@ -50,7 +54,8 @@ async function getListComment( issueComments ) {
  */
 async function getIssueReferences( octokit, owner, repo, number, issueComments ) {
 	const ticketReferences = [];
-	const referencesRegexP = /[0-9]*-(?:zen|zd)|[a-zA-Z0-9-]+-p2#comment-[0-9]*/gim;
+	const referencesRegexP =
+		/[0-9]*-(?:zen|zd)|[a-zA-Z0-9-]+-p2#comment-[0-9]*|https:\/\/wordpress\.com\/(?:[a-z]+\/)?forums\/topic\/([a-zA-Z0-9-]+)/gim;
```
Comment (anomiex) [inline projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js]: May as well make the last bit non-capturing too
```suggestion
		/[0-9]*-(?:zen|zd)|[a-zA-Z0-9-]+-p2#comment-[0-9]*|https:\/\/wordpress\.com\/(?:[a-z]+\/)?forums\/topic\/(?:[a-zA-Z0-9-]+)/gim;
```

Comment (Aurorum) [reply projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js]: 👍 done!

**Review by anomiex — COMMENTED** [review]
I wonder whether this bit should be updated
https://github.com/Automattic/jetpack/blob/e19bcdb1d0bb38799337907c784008920fc5467f/projects/github-actions/repo-gardening/src/tasks/gather-support-references/index.js#L95-L98

Glancing at the existing forum threads it seems like the URLs are already all-lowercase, but it still seems wrong to be lowercasing URLs.

**Comment by Aurorum** [comment]
Sure, that make sense. I've added that now - can be seen here: https://github.com/Aurorum/jetpack/issues/5#issuecomment-1902098432

**Review by jeherve — APPROVED** [review]

**Pull request merged**